### ISSUE TITLE
[MegaMek 0.47.0 Bug] getStartingCoordsArray Error

### ISSUE THREAD

**Issue by SirMegaV** [MegaMek/megamek#1500]
### Environment

MegaMek 0.47.0

### Description

Bot controlled warship cannot deploy? Error getStartingCoordsArray.

### Files

Log and save game attached.

[getStartingCoordsArray-megameklog.txt](https://github.com/MegaMek/megamek/files/3645185/getStartingCoordsArray-megameklog.txt)
[getStartingCoordsArray-Round_00.sav.gz](https://github.com/MegaMek/megamek/files/3645186/getStartingCoordsArray-Round_00.sav.gz)

**Comment by sixlettervariables** [comment]
Exception from the log:
```
java.lang.NullPointerException
	at megamek.client.bot.BotClient.getStartingCoordsArray(BotClient.java:879)
	at megamek.client.bot.princess.Princess.calculateDeployment(Princess.java:458)
	at megamek.client.bot.BotClient.calculateMyTurn(BotClient.java:515)
	at megamek.client.bot.BotClient.access$000(BotClient.java:83)
	at megamek.client.bot.BotClient$CalculateBotTurn.run(BotClient.java:104)
```

**Comment by sixlettervariables** [comment]
This should be fixed in #1504 which will be in the next dev release. I was unable to exactly replicate your exception, but did identify two possibilities for the NRE.
